api/query/search.go

```diff
@@ -99,6 +99,25 @@ func (sh structuredSearchHandler) ServeHTTP(w http.ResponseWriter, r *http.Reque
 			val, _ := shared.ParseBooleanParam(q, param)
 			isSimpleQ = isSimpleQ && (val == nil || !*val)
 		}
+		r2 := *r
```

Comment:
you should use request [clone()](https://cs.opensource.google/go/go/+/refs/tags/go1.18.4:src/net/http/request.go;l=372)

Reply:
Would this be correct?
`r2 = r.Clone(r.Context())`

Reply:
That's right!